cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/writer/BasicLineageWriter.java

```diff
@@ -58,9 +59,23 @@ public void addAccess(Id.Run run, Id.DatasetInstance datasetInstance, AccessType
     lineageStore.addAccess(run, datasetInstance, accessType, metadata, component);
   }
 
-  private String gatherMetadata(Id.Run run, Id.DatasetInstance datasetInstance) {
+  @Override
+  public void addAccess(Id.Run run, Id.Stream stream, AccessType accessType) {
+    addAccess(run, stream, accessType, null);
+  }
+
+  @Override
+  public void addAccess(Id.Run run, Id.Stream stream, AccessType accessType, @Nullable Id.NamespacedId component) {
+    // TODO: avoid duplicate writes for a stream instance
+    String metadata = gatherMetadata(run, stream);
+    LOG.debug("Writing access for run {}, stream {}, accessType {}, component {}, metadata = {}",
+              run, stream, accessType, component, metadata);
+    lineageStore.addAccess(run, stream, accessType, metadata, component);
+  }
+
+  private String gatherMetadata(Id.Run run, Id.NamespacedId id) {
```

Comment:
I thought about that, but decided there is no strong case to introduce Id.Data now. If we encounter more such case, then let's do it.